main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/sdf1979/go_final_project/api"
+	"github.com/sdf1979/go_final_project/db"
+	"github.com/sdf1979/go_final_project/signin"
+)
+
+func getPort() int {
+	port := 7540
+	envPort := os.Getenv("TODO_PORT")
+	if len(envPort) > 0 {
+		if eport, err := strconv.ParseInt(envPort, 10, 32); err == nil {
+			port = int(eport)
+		}
+	}
+	return port
+}
+
+func getFileDb() string {
+	appPath, err := os.Executable()
+	if err != nil {
+		log.Fatal(err)
+	}
+	dbFile := filepath.Join(filepath.Dir(appPath), "scheduler.db")
+	envDbFile := os.Getenv("TODO_DBFILE")
+	if len(envDbFile) > 0 {
+		dbFile = envDbFile
+	}
+	return dbFile
+}
+
+func main() {
+
+	dbFile := getFileDb()
+	store, err := db.Open(dbFile)
+	if err != nil {
+		panic(err)
+	}
+	api.SetStore(store)
+
+	defer store.Close()
+
+	port := getPort()
+	webDir := "./web"
+
+	mux := http.NewServeMux()
+	mux.Handle("/", http.FileServer(http.Dir(webDir)))
+	mux.HandleFunc("/api/nextdate", api.NextDateHandler)
+	mux.HandleFunc("/api/task", signin.Auth(api.TaskHandler))
```

Comment:
Тут при желании можно сделать так
```suggestion
	mux.HandleFunc("/api/task", signin.Auth(api.TaskHandler(store)))
```
Тогда  `store` мы гарантировано передадим инициализированный стор и не надо проверять, а внутри так
```golang
func TaskHandler(store db.Store) func(writer http.ResponseWriter, request *http.Request) {
	return func(writer http.ResponseWriter, request *http.Request) {
		switch request.Method {
		case http.MethodPost:
			taskPost(writer, request, store)
		case http.MethodGet:
			taskGet(writer, request, store)
		case "PUT":
			taskPut(writer, request, store)
		case "DELETE":
			taskDelete(writer, request, store)
		default:
			http.Error(writer, http.StatusText(http.StatusMethodNotAllowed), http.StatusMethodNotAllowed)
		}
	}
}
```

Reply:
Ну и глобальный стор убрать